### PR TITLE
add JobFailedValidation reason constant

### PR DESCRIPTION
**Pull request by HeGaoYuan**
add JobFailedValidation  referring to https://github.com/kubeflow/training-operator/pull/1705

### CODE REVIEW

File: pkg/util/status.go [hunk 1]
```diff
@@ -15,8 +15,10 @@ const (
 	JobRunningReason = "JobRunning"
 	// JobFailedReason is added in a job when it is failed.
 	JobFailedReason = "JobFailed"
-	// JobRestarting is added in a job when it is restarting.
+	// JobRestartingReason is added in a job when it is restarting.
 	JobRestartingReason = "JobRestarting"
+	// JobFailedValidationReason is added in a job when it failed validation
+	JobFailedValidationReason = "JobFailedValidation"
```
Comment (johnugeorge) [inline pkg/util/status.go:21]: Do we need this? Why not use `JobFailedReason` and put the right message?

Comment (HeGaoYuan) [reply pkg/util/status.go:21]: It depends on whether we set job to failed when we failed validation.
By the way, the state transition table of job status is not clear now.

**Comment by johnugeorge** [comment]
@tenzen-y  @terrytangyuan 
Should we merge this in this release ? 
Ref: https://github.com/kubeflow/training-operator/pull/1705#issuecomment-1377546403

File: pkg/util/status.go (hunk 1 above)
Comment (tenzen-y) [inline pkg/util/status.go:21]: Once we introduce CEL validation, we will not use that status.
So, can you add a `TODO` comment with https://github.com/kubeflow/training-operator/issues/1708?

**Comment by tenzen-y** [comment]
> @tenzen-y @terrytangyuan Should we merge this in this release ? Ref: [kubeflow/training-operator#1705 (comment)](https://github.com/kubeflow/training-operator/pull/1705#issuecomment-1377546403)

Yes, this status helps users to find a reason for errors.

**Review by terrytangyuan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**